### DISCUSSION TITLE
problem(s) compling COS-Tools and Ack

### DISCUSSION THREAD

**Discussion by nobunagashino** [kej715/COS-Tools#4] [Q&A]
Hi,
When I run make in COS-Tools I get
make -C cos-interface ; \
make -C cos-commands ; \
make -C fortran
make[1]: Entering directory '/home/ian/COS-Tools/cos-interface'
../cal -t systxt args.cal
../cal -t systxt dmp.cal
../cal -t systxt iutil.cal
../cal -t systxt pack.cal
../cal -t systxt sysio.cal
../cal -t systxt unpack.cal
rm -f libcos.lib ; ../lib -o libcos.lib args.obj dmp.obj iutil.obj pack.obj sysio.obj unpack.obj
make[1]: Leaving directory '/home/ian/COS-Tools/cos-interface'
make[1]: Entering directory '/home/ian/COS-Tools/cos-commands'
../ldr -o charges.abs ../cos-interface/libcos.lib charges.obj
make[1]: *** [Makefile:30: charges.abs] Segmentation fault (core dumped)
make[1]: Leaving directory '/home/ian/COS-Tools/cos-commands'
make[1]: Entering directory '/home/ian/COS-Tools/fortran'
cc  -o kftc binops.o codegen.o coercion.o compile.o fnv32a.o global.o lex.o list.o main.o services.o symbols.o 
/usr/bin/ld: binops.o: in function `cstExpInt':
/home/ian/COS-Tools/fortran/binops.c:67:(.text+0x14d): undefined reference to `pow'
/usr/bin/ld: binops.o: in function `cstExpReal':
/home/ian/COS-Tools/fortran/binops.c:73:(.text+0x198): undefined reference to `pow'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:60: kftc] Error 1
make[1]: Leaving directory '/home/ian/COS-Tools/fortran'
make: *** [Makefile:92: all] Error 2

The Pow  reference can be resolved by linking -lm in the fortran directory but the segmentation fault looks more serious-if I run hello.sh in the examples directory it also dumps core without creating an executable so something may be broken..

Ack also has problems compiling-it hangs on lang/m2/comp+next_c 
This can be resolved by editing /ack/lang/m2/comp/build.lua  and deleting line 69.

I am using Ubuntu 24.04 but have also tried redhat 8 and debian 12.7 with the same results.

Any help gratefully received but will be aboard soon without access to a PC so it might be a while for me to respond.

Going to be abroad shortly without access to

**Comment by kej715** [comment]
Thanks for reporting the issues and providing suggested corrections. Code
has been committed to resolve the problems.

thanks again,
Kevin
